hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/DistributionSetAssignmentResult.java

```diff
@@ -23,39 +19,29 @@
  */
 public class DistributionSetAssignmentResult extends AssignmentResult<Target> {
 
-    private final List<String> assignedTargets;
-    private final List<Action> actions;
-
+    private final List<? extends Action> assignedActions;
     private final DistributionSet distributionSet;
 
-    private final TargetManagement targetManagement;
-
     /**
      *
      * Constructor.
-     *
-     * @param distributionSet
+     *  @param distributionSet
      *            that has been assigned
-     * @param assignedTargets
-     *            the target objects which have been assigned to the
-     *            distribution set
-     * @param assigned
-     *            count of the assigned targets
      * @param alreadyAssigned
-     *            the count of the already assigned targets
-     * @param actions
-     *            of the assignment
-     * @param targetManagement
-     *            to retrieve the assigned targets
+     *            the the count of already assigned targets
+     * @param assigned
+ *            the assigned targets
+     * @param unassigned
+*            the unassigned targets
+     * @param assignedActions
+     *       the created Actions as a result of this assignment
      */
-    public DistributionSetAssignmentResult(final DistributionSet distributionSet, final List<String> assignedTargets,
-            final int assigned, final int alreadyAssigned, final List<Action> actions,
-            final TargetManagement targetManagement) {
-        super(assigned, alreadyAssigned, 0, Collections.emptyList(), Collections.emptyList());
+    public DistributionSetAssignmentResult(final DistributionSet distributionSet, final int alreadyAssigned,
+            final List<? extends Target> assigned, final List<? extends Target> unassigned,
+            final List<? extends Action> assignedActions) {
+        super(alreadyAssigned, assigned, unassigned);
```

Comment:
Within an DistributionSetAssignment we will never unassign Targets, so I think the parameter can be removed and an empty list can be passed to the super constructor like it was done before.